core/tests/DesktopTests/TestInfrastructure.Appium/TestsBase.cs

```diff
@@ -0,0 +1,111 @@
+
+        protected Logger Logger { get; private set; }
+
+        protected DeviceSession DeviceSession { get; private set; }
+
+        /// <summary>
+        /// MSTest will create this object.
+        /// </summary>
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            Logger = new Logger(TestContext);
+            Logger.LogInfo($"Starting test {TestContext.TestName}");
+
+            InitSession();
+        }
+
+        [TestCleanup]
+        public virtual void TestCleanup()
+        {
+            try
+            {
+                Logger.LogInfo($"Entering Test Cleanup. Outcome: {TestContext.CurrentTestOutcome}");
+                if (DeviceSession != null)
+                {
+                    if (TestContext.CurrentTestOutcome != UnitTestOutcome.Passed)
+                    {
+                        Logger.LogInfo("Taking failure screenshot");
+                        DeviceSession.TakeScreenshot("Failure");
+                    }
+                }
+            }
+            finally
+            {
+                if (DeviceSession != null)
+                {
+                    DeviceSession.Dispose();
+                    TrySaveOlympusLogs(DeviceSession.OlympusLogUri);
+                }
+            }
+
+            Logger.LogInfo("Test cleanup finished.");
+        }
+
+        private void InitSession()
+        {
+            Assert.IsNotNull(Logger, "Initialize the logger before calling this method");
+            try
+            {
+                DeviceSession = new DeviceSession(Logger, PlatformType);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError("Failed to initialize the driver.", ex);
+                throw;
+            }
+        }
+
+        private void TrySaveOlympusLogs(Uri uri)
+        {
+            // Give Olympus a chance!
+            Thread.Sleep(1000);
+            try
+            {
+                using (var client = new HttpClient())
+                {
+                    Logger.LogInfo($"Attempting to fetch Olympus logs from {uri}");
+
+                    var downloadTask = client.GetAsync(uri);
+                    downloadTask.Wait();
+                    HttpResponseMessage response = downloadTask.Result;
+
+                    if (response.IsSuccessStatusCode)
+                    {
+                        var contentTask = response.Content.ReadAsStringAsync();
+                        string content = contentTask.Result;
+                        Logger.AddContentAsLogFile(content, "Olympus.log");
+                    }
+                    else if (response.StatusCode == HttpStatusCode.NotFound)
+                    {
+                        // This might be the case when not pointing at Olympus or if the session init failed.
+                        Logger.LogInfo("Olympus logs do not exist for this session.");
```

Comment:
>Olympus [](start = 40, length = 7)

lots of mentioning of Olympus everwhere. I assume it's not relevant?

Reply:
Yes, the Olympus stuff are just remnants of the old infrastructure. will be removed